File: backend/src/main/kotlin/com/logvue/data/model/Timeline.kt
package com.logvue.data.model

import kotlinx.serialization.Serializable

@Serializable
data class TimelineRequest(
    val fileId: String,
    val resolution: String = "sec"
)

@Serializable
data class TimelineResponse(
    val timeRange: TimeRange,
    val resolution: String,
    val buckets: List<TimelineBucket>,
    val tagColors: Map<String, String>
)

@Serializable
data class TimelineBucket(
    val timestamp: Long,
    val count: Int,
    val tags: Map<String, Int>
)

// TimeRange is already defined in LogFile.kt in the same package

File: backend/models/timeline.go
package models

// TimelineRequest is the request for timeline buckets
type TimelineRequest struct {
	FileID      string `json:"fileId"`
	NumBuckets  *int   `json:"numBuckets,omitempty"`
	TimeFrom    *int64 `json:"timeFrom,omitempty"`
	TimeTo      *int64 `json:"timeTo,omitempty"`
}

// TimelineBucket represents a single time bucket
type TimelineBucket struct {
	Timestamp int64          `json:"timestamp"`
	Count     int            `json:"count"`
	Tags      map[string]int `json:"tags"`
}

// TimelineResponse is the timeline with buckets and colors
type TimelineResponse struct {
	TimeRange TimeRange            `json:"timeRange"`
	Buckets   []TimelineBucket     `json:"buckets"`
	TagColors map[string]string   `json:"tagColors"`
}
